src/main/java/com/example/forecasting/datasource/PostgresDataSourceLoader.java

```diff
@@ -0,0 +1,26 @@
+package com.example.forecasting.datasource;
+
+import com.zaxxer.hikari.HikariDataSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.stereotype.Component;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+@Component
+public class PostgresDataSourceLoader implements ApplicationRunner {
+    private final HikariDataSource hikariDataSource;
+
+    @Autowired
+    public PostgresDataSourceLoader(HikariDataSource hikariDataSource) {
+        this.hikariDataSource = hikariDataSource;
+    }
+
+    @Override
+    public void run(ApplicationArguments args) throws SQLException {
+        Connection conn = hikariDataSource.getConnection();
```

Comment:
This will currently not allow spring boot application to boot up if the DB is not running. instead catch and throw a warning maybe.